versions.props

```diff
@@ -0,0 +1,23 @@
+org.slf4j:slf4j-api = 1.7.5
+com.google.guava:guava = 28.0-jre
+org.apache.avro:avro = 1.8.2
+org.apache.hadoop:* = 2.7.3
+org.apache.hive:hive-standalone-metastore = 1.2.1
+org.apache.orc:orc-core = 1.5.5
+org.apache.parquet:parquet-avro = 1.10.0
+org.apache.spark:spark-hive_2.11 = 2.4.0
+org.apache.pig:pig = 0.14.0
+com.fasterxml.jackson.core:jackson-core = 2.7.9
```

Comment:
Before this change, these Jackson versions used to be inconsistent. We were declaring that we wanted Jackson 2.6.7. However, spark-sql always pulled in the transitive dependency of jackson-core at version 2.7.9. I discovered this by looking at the rendered `versions.lock` file below, using `./gradlew why` to trace back the source of the dependency, and also verified using `./gradlew dependencyInsight...` before this diff.

I think it's safest to fix all the Jackson components at the same version, in which case I thought we should bias towards the newer version.

Reply:
Nice spot!
Worth nothing that in GCV land, these lines don't "fix" the version but instead provide a required constraint, rather than a force, meaning gradle can choose a higher version if other constraints come into play, so all constraints, even from transitives, are always respected. 
However, since they are independent lines, the individual jackson modules might still end up with different versions later on as you upgrade other dependencies.

If you want to ensure all jackson modules use the same version, you should use a star-matcher here:
```
com.fasterxml.jackson.*:* = 2.7.9
```

Under the scenes, this creates a virtual platform that ensures all jackson modules' versions align to the highest version, whatever that ends up being.

This way, if some transitive ends up requiring jackson-core:2.9.8 for instance, then _all_ jackson versions end up being bumped to 2.9.8, thus ensuring they've been tested with each other.

Check out gradle's [example of using virtual platforms](https://docs.gradle.org/current/userguide/managing_transitive_dependencies.html#aligning_versions_of_modules_not_published_with_gradle) which deals with exactly this problem.

Reply:
Good call - I changed the module label to use wildcards.